features/steps/jobs_steps.go

```diff
@@ -121,22 +120,29 @@ func (f *JobsFeature) RegisterSteps(ctx *godog.ScenarioContext) {
 }
 
 // Reset sets the resources within a specific JobsFeature back to their default values.
-func (f *JobsFeature) Reset(mongoFail bool) *JobsFeature {
+func (f *JobsFeature) Reset(mongoFail bool) error {
+	var err error
 	if mongoFail {
 		f.MongoClient.Database = "lost database connection"
 	} else {
 		f.MongoClient.Database = memongo.RandomDatabase()
 	}
 	ctx := context.Background()
-	cfg, err := config.Get()
-	if err != nil {
-		return nil
+	if f.Config == nil {
+		cfg, err := config.Get()
+		if err != nil {
+			return err
```

Comment:
So as I talked about in my presentation about errors I think now's a good a time as any to start encouraging people.. Pretty much all the time instead of just returning an error it's better to add a small amount of context describing what the action was that just failed. I.E. here  `return err` would become `return fmt.Errorf("failed to get config: %w", err)` and below would become `return fmt.Errorf("failed to initialise Mongo client: %w"). If we do this each and every time we return an error we end up with an error which is really informative and helps us debug much more efficiently.

Reply:
done, thanks for explaining how to do this